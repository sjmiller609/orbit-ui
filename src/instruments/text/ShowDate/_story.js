import React from 'react'
import { storiesOf } from '@storybook/react'
import { ShowDate } from 'instruments'

storiesOf('Instruments|Text', module).add('ShowDate', () => (
  <ShowDate date={new Date()} />
))
