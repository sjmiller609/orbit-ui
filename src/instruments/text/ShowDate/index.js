'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'

//TODO: add formatting options
const ShowDate = ({ date, className, seconds }) => {
  if (!date) return null
  const d = new Date(date)
  const now = new Date()
  let today
  let options = {
    month: 'long',
    day: 'numeric',
  }

  // show year if different
  if (d.getFullYear() !== now.getFullYear()) options.year = 'numeric'
  else {
    if (d.getMonth() === now.getMonth()) {
      // today
      if (d.getUTCDate() === now.getUTCDate()) {
        today = 'Today, '
        options = {
          hour: 'numeric',
          minute: 'numeric',
        }
        // this month
      } else options.weekday = 'short'
    }
  }

  if (seconds) {
    options.hour = 'numeric'
    options.minute = 'numeric'
    options.second = 'numeric'
  }

  const date2 = !today
    ? d.toLocaleDateString('en-US', options)
    : d.toLocaleTimeString('en-US', options)

  return (
    <span className={classnames(s.date, className)}>
      {today}
      {date2}
    </span>
  )
}

ShowDate.propTypes = {
  date: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
  weekday: PropTypes.bool,
  seconds: PropTypes.bool,
  className: PropTypes.string,
}

export default ShowDate
