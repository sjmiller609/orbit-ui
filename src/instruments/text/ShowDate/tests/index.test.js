import React from 'react'
import { shallow } from 'enzyme'
import ShowDate from '../index'

const props = {
  date: '2019-08-05T04:41:20',
  weekday: true,
  seconds: false,
  className: 'testClassString',
}

describe('ShowDate', () => {
  it('should be defined', () => {
    expect(ShowDate).toBeDefined()
  })

  it('renders correctly', () => {
    const wrapper = shallow(<ShowDate {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('handles no date', () => {
    const wrapper = shallow(<ShowDate {...props} date={null} />)
    expect(wrapper).toMatchSnapshot()
  })
})
