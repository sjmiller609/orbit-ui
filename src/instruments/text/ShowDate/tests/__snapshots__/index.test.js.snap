// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowDate handles no date 1`] = `""`;

exports[`ShowDate renders correctly 1`] = `
<span
  className="date testClassString"
>
  August 5
</span>
`;
